src/MoBi.Presentation/Presenter/Main/SimulationExplorerPresenter.cs

```diff
@@ -188,21 +188,51 @@ private void refreshDisplayedSimulation(IMoBiSimulation simulation)
          var simulationNode = _view.NodeById(simulation.Id);
          
          var changedTemplateBuildingBlocks = findChangedTemplateBuildingBlocks(simulation).ToList();
+         var changedModules = findChangedTemplateModules(simulation).ToList();
 
          var isChangedSimulation = changedTemplateBuildingBlocks.Any() || simulation.OriginalQuantityValues.Any();
          simulationNode.Icon = isChangedSimulation ? ApplicationIcons.SimulationRed : ApplicationIcons.SimulationGreen;
 
+         updateModuleIcons(changedModules.Select(x => x.simulationModule), simulationNode.AllNodes.OfType<ModuleConfigurationNode>());
          updateBuildingBlockIcons(changedTemplateBuildingBlocks.Select(x => x.simulationBuildingBlock), simulationNode.AllNodes.OfType<BuildingBlockNode>());
       }
 
+      private IEnumerable<(Module templateModule, Module simulationModule)> findChangedTemplateModules(IMoBiSimulation simulation)
+      {
+         return simulation.Configuration.ModuleConfigurations.Select(x => (templateModule: _interactionTasksForSimulation.TemplateModuleFor(x.Module), moduleConfiguration: x)).
+            Where(match => !versionMatch(match)).Select(x => (templateModule: x.templateModule, simulationModule: x.moduleConfiguration.Module));
+      }
+
+      private bool versionMatch((Module templateModule, ModuleConfiguration moduleConfiguration) match)
+      {
+         var moduleConfiguration = match.moduleConfiguration;
+         var simulationModule = moduleConfiguration.Module;
+         var templateInitialConditions = moduleConfiguration.SelectedInitialConditions == null ? null : _interactionTasksForSimulation.TemplateBuildingBlockFor(moduleConfiguration.SelectedInitialConditions);
```

Comment:
if TemplateBuildingBlockFor returns null if the passed parameter is null, then you don't nee dall those condition all the time. Woudl iut make sense?